### DISCUSSION TITLE
Thoughts about upgrading TypeScript version used by PixiJS

### DISCUSSION THREAD

**Discussion by SuperSodaSea** [pixijs/pixijs#9116] [Ideas]
Upgrading to a higher TypeScript version can be useful, we can make use of new features and new types of TypeScript.

For example, some type definitions are only available in higher version of TypeScript:

- `WEBGL_compressed_texture_etc/etc1/pvrtc` (Appears in the exported types)
- `Intl.Segmenter` (Internal use only)

With TypeScript 4.5+ we can use *lib replacement* to use a newer version of `lib.dom.d.ts` without upgrading the version of TypeScript itself, see [@types/web](https://www.npmjs.com/package/@types/web) for details. Sadly there seems to be no such package for types outside of `lib.dom.d.ts` (e.g. `Intl.Segmenter`).

Upgrading to a higher version of TypeScript should be very easy, but I am concerned that this may cause some compatibility issues for users. What do you guys think?

**Comment by bigtimebuddy** [comment]
@Zyie could provide more context here but I remember we were blocked in upgrading which is why it hasn't moved yet.

**Reply by Zyie** [reply]
I believe we were blocked by issues with `@microsoft/api-extractor`, however we no longer use this since we moved away from bundling everything into a single file

So we should be ok to upgrade typescript again

**Reply by miltoncandelero** [reply]
API extractor works with latest ts anyway 😅

**Comment by miltoncandelero** [comment]
From the new typescript I loved the feature that forces you to say override when you override, prevents silly mistakes when making methods and the import types as values thingy set to error to allow further decoupling of code.

I am all in for the latest typescript we can get, however with ts5 in the horizon maybe wait for that to release?

Also also, we compile with esbuild, so some features of typescript fly out the window during compilation 🤔

**Reply by SuperSodaSea** [reply]
My only concern is that if the output `.d.ts` contains types that are only available in higher TypeScript versions, users may need to upgrade their TypeScript version as well?

**Reply by bigtimebuddy** [reply]
Relevant TS v5 post https://devblogs.microsoft.com/typescript/announcing-typescript-5-0-beta/
